src/HexColorValidator.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace ipl\Validator;
+
+use ipl\I18n\Translation;
+
+/**
+ * Validator for color input controls
+ */
+class HexColorValidator extends BaseValidator
+{
+    use Translation;
+
+    /**
+     * Check whether the given color is valid
+     *
+     * @param string $value
+     *
+     * @return bool
+     */
+    public function isValid($value): bool
+    {
+        // Multiple isValid() calls must not stack validation messages
+        $this->clearMessages();
+
+        if (! is_string($value)) {
```

Comment:
This is not a validation but a programming error and since `preg_match` already expects a string, you can remove this.